source/Projectile.cpp

```diff
@@ -76,8 +76,11 @@ Projectile::Projectile(const Projectile &parent, const Weapon *weapon)
 		if(!parent.weapon->Acceleration())
 		{
 			// Move in this new direction at the same velocity.
-			double parentVelocity = parent.weapon->Velocity();
-			velocity += (this->angle.Unit() - parent.angle.Unit()) * parentVelocity;
+			// Velocity may be negative.
+			Point referenceVector = parent.angle.Unit();
+			double parentVelocity = parent.velocity.Y() * referenceVector.Y()
+				+ parent.velocity.X() * referenceVector.X();
```

Comment:
I'm curious as to why this needs to be done to get the parent velocity. Does parent.velocity.Length() not return the same value? If not, then what's the difference?

Reply:
Length() results an absolute value and it cannot decide whether a velocity is positive or negative.
I'm not sure that I may omit handling a negative velocity.

This calculates the part of parallel to the reference vector of the parent's velocity, that is, the signed value of the parent's velocity.

Reply:
That’s what I thought. Maybe clarify the comment more beyond “Velocity can be negative.”

“To maintain the sign of the velocity, Point::Length can’t be used” or something like that.